# Phase 06 — HDC Variant Screening

This executed Notebook initializes the Phase 06 interface and validates frozen upstream evidence. It does not train, tune, select, or evaluate a model.

Required P1 variants:

1. Vanilla Prototype HDC
2. OnlineHD-style HDC
3. Multi-centroid HDC
4. HDC+OnlineHD Hybrid

Vanilla Prototype HDC is reused read-only from frozen Phase 05. The regression task is **bounded difficulty-induced workload proxy regression**, and project conclusions are limited to **workload-proxy classification and regression**. The target is not directly measured continuous cognitive workload.

## 1. Project root and read-only preflight

In [1]:
from pathlib import Path
import json
import sys

phase_dir = Path.cwd().resolve()
assert phase_dir.name == "phase_06_hdc_variant_screening"
project_root = phase_dir.parents[1]
assert (project_root / "最新完整实验计划_分类回归双任务.md").is_file()
assert (project_root / "CODEX_NOTEBOOK_RULES.md").is_file()
sys.path.insert(0, str(phase_dir / "src"))

from phase06_preflight import run_preflight

summary = run_preflight(phase_dir)
print(f"PROJECT ROOT: {project_root}")
print("UPSTREAM ACCESS MODE: READ ONLY")
print("PRIMARY DATA OR FOLDS COPIED: NO")

PROJECT ROOT: E:\hdc-vr-pilot
UPSTREAM ACCESS MODE: READ ONLY
PRIMARY DATA OR FOLDS COPIED: NO


## 2. Primary data and frozen-fold evidence

In [2]:
print(f"PRIMARY DATA ROWS: {summary['modeling_rows']}")
print(f"SUBJECTS: {summary['subjects']}")
print(f"PRIMARY FEATURES: {summary['primary_predictive_features']}")
print(f"UNIQUE RUN KEYS: {summary['unique_run_keys']}")
print(f"TARGET_CLASS VALUES: {summary['target_class_values']}")
print(f"TARGET_CLASS MISSING: {summary['target_class_missing']}")
print(f"TARGET_SCORE VALUES: {summary['target_score_values']}")
print(f"TARGET_SCORE MISSING: {summary['target_score_missing']}")
print(f"PRIMARY DATA SHA-256: {summary['primary_sha256']}")
print(f"PRIMARY DATA CHECKSUM: {summary['primary_checksum']}")
print(f"FROZEN FOLD SHA-256: {summary['frozen_fold_sha256']}")
print(f"FROZEN FOLD CHECKSUM: {summary['frozen_fold_checksum']}")
print(f"FOLD ROWS / UNIQUE RUNS: {summary['fold_assignment_rows']} / {summary['fold_unique_run_keys']}")
print(f"OUTER FOLDS: {summary['outer_folds']}")
print(f"OUTER SUBJECT ISOLATION: {summary['outer_subject_isolation']}")
print(f"INNER 3-FOLD GROUPKFOLD FEASIBILITY: {summary['inner_groupkfold_3_feasibility']}")

PRIMARY DATA ROWS: 419
SUBJECTS: 35
PRIMARY FEATURES: 1176
UNIQUE RUN KEYS: 419
TARGET_CLASS VALUES: [0, 1, 2, 3]
TARGET_CLASS MISSING: 0
TARGET_SCORE VALUES: [1.0, 2.0, 3.0, 4.0]
TARGET_SCORE MISSING: 0
PRIMARY DATA SHA-256: 0a2aef89c01b43c3a4e5afe40b96797784627665490982731b2541f96b45fc44
PRIMARY DATA CHECKSUM: PASS
FROZEN FOLD SHA-256: e4dc943af21851bace49345f6336f9c88f82613ca3a26b3c433efe7dfb041f6f
FROZEN FOLD CHECKSUM: PASS
FOLD ROWS / UNIQUE RUNS: 419 / 419
OUTER FOLDS: 5
OUTER SUBJECT ISOLATION: PASS
INNER 3-FOLD GROUPKFOLD FEASIBILITY: PASS


## 3. Frozen Phase 05 Vanilla HDC interface

In [3]:
print(f"PHASE 05 STATUS: {summary['phase05']['status']}")
print(f"PHASE 05 FREEZE INTERFACE: {summary['phase05']['freeze_interface']}")
print(f"PHASE 05 VANILLA RESULTS AVAILABLE: {summary['phase05']['vanilla_results_available']}")
print("PHASE 05 VANILLA BASELINE ACCESS: READ-ONLY REUSE; RETRAINING PROHIBITED")
print(f"PHASE 05 CANONICAL CONFIGURATION SELECTION: {summary['phase05']['canonical_configuration_selection']}")
print("PHASE 05 OUTER-TEST OBSERVED BEST DIMENSION/SEED USED FOR PHASE 06 SELECTION: NO")

PHASE 05 STATUS: FROZEN
PHASE 05 FREEZE INTERFACE: PASS
PHASE 05 VANILLA RESULTS AVAILABLE: YES
PHASE 05 VANILLA BASELINE ACCESS: READ-ONLY REUSE; RETRAINING PROHIBITED
PHASE 05 CANONICAL CONFIGURATION SELECTION: NOT_PERFORMED_OUTER_TEST_SELECTION_PROHIBITED
PHASE 05 OUTER-TEST OBSERVED BEST DIMENSION/SEED USED FOR PHASE 06 SELECTION: NO


## 4. Frozen Phase 04 comparison interfaces

In [4]:
print(f"PHASE 04A INTERFACE: {summary['phase04']['phase04a']}")
print(f"PHASE 04B INTERFACE: {summary['phase04']['phase04b']}")
print("TRADITIONAL MODEL RESULTS RECOMPUTED: NO")

PHASE 04A INTERFACE: PASS
PHASE 04B INTERFACE: PASS
TRADITIONAL MODEL RESULTS RECOMPUTED: NO


## 5. Initialization audit readback

In [5]:
audit_paths = {
    "INPUT AND FOLD AUDIT": phase_dir / "audits" / "phase06_input_and_fold_audit.json",
    "PHASE 05 FREEZE INTERFACE AUDIT": phase_dir / "audits" / "phase06_phase05_freeze_interface_audit.json",
    "PHASE 04 BASELINE INTERFACE AUDIT": phase_dir / "audits" / "phase06_phase04_baseline_interface_audit.json",
    "INITIALIZATION ARTIFACT AUDIT": phase_dir / "audits" / "phase06_initialization_artifact_audit.json",
}
for label, path in audit_paths.items():
    payload = json.loads(path.read_text(encoding="utf-8"))
    print(f"{label}: {payload['result']}")
print(f"INPUT MANIFEST SAVED: {summary['input_manifest_saved']}")

INPUT AND FOLD AUDIT: PASS
PHASE 05 FREEZE INTERFACE AUDIT: PASS
PHASE 04 BASELINE INTERFACE AUDIT: PASS
INITIALIZATION ARTIFACT AUDIT: PASS
INPUT MANIFEST SAVED: YES


## Phase Validation Summary

VERIFIED: actual Primary data, frozen folds, outer subject isolation, inner GroupKFold feasibility, frozen Phase 04A/04B interfaces, and frozen Phase 05 Vanilla HDC interface.

NOT VERIFIED: Phase 06 algorithm definitions and model-selection rules remain intentionally deferred to contract freeze.

WARNINGS: no Phase 05 outer-test observed best dimension or seed may be promoted to a canonical Phase 06 configuration.

KEY RESULTS: initialization evidence only; no model result was generated.

OUTPUT FILES: Phase 06 configs, manifest, audits, README, source validator, tests, and this executed Notebook.

NEXT PHASE REQUIREMENTS: complete and freeze the Phase 06 modeling contract before any HDC variant modeling.

In [6]:
gates = [
    summary["primary_checksum"] == "PASS",
    summary["frozen_fold_checksum"] == "PASS",
    summary["outer_subject_isolation"] == "PASS",
    summary["inner_groupkfold_3_feasibility"] == "PASS",
    summary["phase05"]["freeze_interface"] == "PASS",
    summary["phase04"]["phase04a"] == "PASS",
    summary["phase04"]["phase04b"] == "PASS",
    summary["initialization_audit"] == "PASS",
]
status = "PENDING_CONTRACT_FREEZE" if all(gates) else "FAIL"
print("HDC VARIANT TRAINING EXECUTED: NO")
print(f"PHASE 06 STATUS: {status}")
print(f"READY FOR PHASE 06 CONTRACT FREEZE: {'YES' if all(gates) else 'NO'}")

HDC VARIANT TRAINING EXECUTED: NO
PHASE 06 STATUS: PENDING_CONTRACT_FREEZE
READY FOR PHASE 06 CONTRACT FREEZE: YES


<!-- phase06-quick-screen-completion-v1 -->
## Phase 06 Contract Freeze and Quick-Screen Completion

This section records classification-only inner-CV quick screening. It does not contain outer-test predictions, regression heads, Final Confirmation, or a final-best-HDC claim.

In [7]:
from pathlib import Path
import json
import pandas as pd

phase_dir = Path.cwd().resolve()
amendment = json.loads((phase_dir / "audits/phase06_phase05_amendment_gate_audit.json").read_text(encoding="utf-8"))
contract = json.loads((phase_dir / "configs/phase06_hdc_variant_contract.json").read_text(encoding="utf-8"))
spaces = json.loads((phase_dir / "configs/phase06_variant_search_spaces.json").read_text(encoding="utf-8"))
contract_audit = json.loads((phase_dir / "audits/phase06_contract_freeze_audit.json").read_text(encoding="utf-8"))
unit_audit = json.loads((phase_dir / "audits/phase06_unit_test_audit.json").read_text(encoding="utf-8"))
print(f"PHASE 05 FREEZE AMENDMENT: {amendment['result']}")
print(f"PHASE 06 CONTRACT FREEZE: {contract_audit['result']}")
print(f"PHASE 06 UNIT TESTS: {unit_audit['result']}")
print("REPRESENTATION / BINDING / BUNDLING / SIMILARITY:", contract["common_interface"]["representation"], "/", contract["common_interface"]["binding"], "/", contract["common_interface"]["bundling"], "/", contract["common_interface"]["similarity"])
print("VARIANTS:", [item["name"] for item in contract["variants"].values()])
print("SEARCH COUNTS:", {name: spaces[name]["total_candidates"] for name in ["onlinehd", "multicentroid", "hybrid"]})

PHASE 05 FREEZE AMENDMENT: PASS
PHASE 06 CONTRACT FREEZE: PASS
PHASE 06 UNIT TESTS: PASS
REPRESENTATION / BINDING / BUNDLING / SIMILARITY: bipolar / bipolar elementwise multiplication / sum then deterministic sign with frozen tie vector / cosine
VARIANTS: ['Vanilla Prototype HDC', 'OnlineHD-style HDC', 'Multi-centroid HDC', 'HDC+OnlineHD Hybrid']
SEARCH COUNTS: {'onlinehd': 24, 'multicentroid': 6, 'hybrid': 32}


In [8]:
all_folds = json.loads((phase_dir / "audits/phase06_quick_screen_all_folds_audit.json").read_text(encoding="utf-8"))
best = pd.read_csv(phase_dir / "results/summaries/phase06_all_variants_quick_screen_summary.csv")
display(best[["variant", "outer_fold", "candidate_id", "dimension", "mean_macro_f1", "std_macro_f1_sample", "mean_balanced_accuracy", "mean_severe_error_rate"]])
for variant in ["onlinehd", "multicentroid", "hybrid"]:
    evidence = all_folds["variants"][variant]
    print(f"{variant.upper()} QUICK-SCREEN FOLDS: {evidence['folds_completed']}/5; CANDIDATES PER FOLD: {evidence['candidates_per_fold'][0]}")
print(f"ALL BEST CONFIGS REPRODUCIBLE: {'PASS' if all_folds['all_best_configs_reproducible'] else 'FAIL'}")

,variant,outer_fold,candidate_id,dimension,mean_macro_f1,std_macro_f1_sample,mean_balanced_accuracy,mean_severe_error_rate
0,onlinehd,1,candidate_001,2000,0.794042,0.096024,0.798784,0.029152
1,onlinehd,2,candidate_009,2000,0.768158,0.051113,0.771743,0.027207
2,onlinehd,3,candidate_002,2000,0.774666,0.042938,0.782189,0.032524
3,onlinehd,4,candidate_024,5000,0.803248,0.099665,0.806008,0.038126
4,onlinehd,5,candidate_018,5000,0.814397,0.059713,0.815827,0.039506
5,multicentroid,1,candidate_004,5000,0.826820,0.116290,0.831013,0.030294
6,multicentroid,2,candidate_005,5000,0.811358,0.076602,0.813335,0.035896
7,multicentroid,3,candidate_001,2000,0.793827,0.026702,0.795616,0.042069
8,multicentroid,4,candidate_002,2000,0.838189,0.037111,0.840063,0.032809
9,multicentroid,5,candidate_005,5000,0.827328,0.035006,0.829099,0.045679


ONLINEHD QUICK-SCREEN FOLDS: 5/5; CANDIDATES PER FOLD: 24
MULTICENTROID QUICK-SCREEN FOLDS: 5/5; CANDIDATES PER FOLD: 6
HYBRID QUICK-SCREEN FOLDS: 5/5; CANDIDATES PER FOLD: 32
ALL BEST CONFIGS REPRODUCIBLE: PASS


In [9]:
efficiency_rows = []
for variant in ["onlinehd", "multicentroid", "hybrid"]:
    for fold in range(1, 6):
        frame = pd.read_csv(phase_dir / f"results/efficiency/{variant}_quick_screen_fold_{fold}_efficiency.csv")
        efficiency_rows.append({
            "variant": variant, "outer_fold": fold, "candidate_count": len(frame),
            "training_seconds_sum": frame["training_seconds"].sum(),
            "inference_seconds_sum": frame["inference_seconds"].sum(),
            "model_bytes_min": frame["model_bytes"].min(), "model_bytes_max": frame["model_bytes"].max(),
        })
display(pd.DataFrame(efficiency_rows))

,variant,outer_fold,candidate_count,training_seconds_sum,inference_seconds_sum,model_bytes_min,model_bytes_max
0,onlinehd,1,24,3.804180,0.116662,32000,80000
1,onlinehd,2,24,3.617186,0.086609,32000,80000
2,onlinehd,3,24,3.596857,0.084492,32000,80000
3,onlinehd,4,24,3.632234,0.089290,32000,80000
4,onlinehd,5,24,3.606082,0.086038,32000,80000
5,multicentroid,1,6,3.926466,0.036137,64000,320000
6,multicentroid,2,6,3.779704,0.034375,64000,320000
7,multicentroid,3,6,3.778382,0.036764,64000,320000
8,multicentroid,4,6,3.777165,0.036097,64000,320000
9,multicentroid,5,6,3.936917,0.036087,64000,320000


In [10]:
print(f"PRIMARY DATA CHECKSUM: {all_folds['primary_checksum']}")
print(f"FROZEN FOLD CHECKSUM: {all_folds['frozen_fold_checksum']}")
print("OUTER SUBJECT ISOLATION: PASS")
print("INNER SUBJECT ISOLATION: PASS")
print("QUICK-SCREEN LEAKAGE AUDIT:", all_folds["result"])
print("QUICK-SCREEN ARTIFACT AUDIT:", all_folds["result"])
print("HISTORICAL PHASE 03-05 ARTIFACTS UNCHANGED:", "PASS" if all_folds["historical_phase03_to_phase05_artifacts_unchanged"] else "FAIL")
print("OUTER-TEST FEATURE ACCESS: NO")
print("OUTER-TEST LABEL ACCESS: NO")
print("OUTER-TEST PREDICTIONS GENERATED: NO")
print("SIMILARITY REGRESSION EXECUTED: NO")
print("RIDGE READOUT EXECUTED: NO")
print("FINAL CONFIRMATION EXECUTED: NO")
print("PHASE 06 STATUS: QUICK_SCREEN_COMPLETE")
print("READY FOR PHASE 06 FINAL CONFIRMATION: YES")

PRIMARY DATA CHECKSUM: PASS
FROZEN FOLD CHECKSUM: PASS
OUTER SUBJECT ISOLATION: PASS
INNER SUBJECT ISOLATION: PASS
QUICK-SCREEN LEAKAGE AUDIT: PASS
QUICK-SCREEN ARTIFACT AUDIT: PASS
HISTORICAL PHASE 03-05 ARTIFACTS UNCHANGED: PASS
OUTER-TEST FEATURE ACCESS: NO
OUTER-TEST LABEL ACCESS: NO
OUTER-TEST PREDICTIONS GENERATED: NO
SIMILARITY REGRESSION EXECUTED: NO
RIDGE READOUT EXECUTED: NO
FINAL CONFIRMATION EXECUTED: NO
PHASE 06 STATUS: QUICK_SCREEN_COMPLETE
READY FOR PHASE 06 FINAL CONFIRMATION: YES


## Final Confirmation — completed execution

This section records the completed strict nested Final Confirmation for OnlineHD, Multi-centroid, and Hybrid across outer folds 1–5, dimensions 1000/2000/5000/10000, and seeds 42–46 (300 fold-config runs).

- Each outer fold reused only its own Quick Screen selected structural parameters.
- `GroupKFold(n_splits=3, groups=subject_id)` independently fit preprocessing, quantization, hypervectors, and variant models on inner training data.
- Temperature was selected exclusively by inner CV from `[0.05, 0.1, 0.2, 0.5, 1.0, 2.0]` using mean bounded MAE, sample-SD MAE, RMSE, then smaller temperature.
- Outer-test features were loaded only after inner selections were fixed; outer-test labels were joined only after predictions were generated. Outer-test data were not used for tuning.
- Regression is described as **bounded difficulty-induced workload proxy regression**, not directly measured continuous cognitive workload.
- Ridge handling is `COMMON_ENCODER_READOUT_BASELINE`: these variants change prototypes/centroids, not sample hypervectors, so variant-specific repeated Ridge fits would be pseudo-replication.
- Primary SHA-256: `0a2aef89c01b43c3a4e5afe40b96797784627665490982731b2541f96b45fc44`.
- Frozen folds SHA-256: `e4dc943af21851bace49345f6336f9c88f82613ca3a26b3c433efe7dfb041f6f`.
- Leakage, coverage, artifact, and checkpoint-integrity audits passed.
- Final HDC selection and Final OOF Consolidation have **not** been executed.

Quick Screen selected structures by variant/fold:

```json
{
  "onlinehd": {
    "1": {
      "epochs": 1,
      "learning_rate": 0.05,
      "margin_threshold": 0.0
    },
    "2": {
      "epochs": 5,
      "learning_rate": 0.05,
      "margin_threshold": 0.0
    },
    "3": {
      "epochs": 1,
      "learning_rate": 0.05,
      "margin_threshold": 0.1
    },
    "4": {
      "epochs": 5,
      "learning_rate": 0.1,
      "margin_threshold": 0.1
    },
    "5": {
      "epochs": 3,
      "learning_rate": 0.05,
      "margin_threshold": 0.1
    }
  },
  "multicentroid": {
    "1": {
      "centroids_per_class": 2
    },
    "2": {
      "centroids_per_class": 3
    },
    "3": {
      "centroids_per_class": 2
    },
    "4": {
      "centroids_per_class": 3
    },
    "5": {
      "centroids_per_class": 3
    }
  },
  "hybrid": {
    "1": {
      "centroids_per_class": 3,
      "epochs": 1,
      "learning_rate": 0.1,
      "margin_threshold": 0.0
    },
    "2": {
      "centroids_per_class": 2,
      "epochs": 3,
      "learning_rate": 0.05,
      "margin_threshold": 0.0
    },
    "3": {
      "centroids_per_class": 2,
      "epochs": 3,
      "learning_rate": 0.05,
      "margin_threshold": 0.0
    },
    "4": {
      "centroids_per_class": 2,
      "epochs": 3,
      "learning_rate": 0.1,
      "margin_threshold": 0.1
    },
    "5": {
      "centroids_per_class": 3,
      "epochs": 1,
      "learning_rate": 0.1,
      "margin_threshold": 0.0
    }
  }
}
```


In [11]:
# phase06_final_confirmation_executed_v1
from pathlib import Path
import json
import pandas as pd

phase = Path.cwd()
summary = pd.read_csv(phase / 'results/summaries/phase06_final_confirmation_execution_summary.csv')
audit = json.loads((phase / 'audits/phase06_final_confirmation_all_folds_audit.json').read_text(encoding='utf-8'))
completion = summary.groupby(['variant', 'outer_fold']).size().unstack(fill_value=0).to_dict(orient='index')
classification = summary.groupby('variant')[['macro_f1', 'balanced_accuracy']].mean().round(6).to_dict(orient='index')
regression = summary.groupby('variant')[['mae_bounded', 'rmse_bounded']].mean().round(6).to_dict(orient='index')
temperature = summary.groupby(['variant', 'selected_temperature']).size().rename('count').reset_index().to_dict(orient='records')
efficiency = summary.groupby('variant')['model_bytes'].agg(['min', 'max', 'mean']).round(2).to_dict(orient='index')
executed_summary = {
    'status': 'FINAL_CONFIRMATION_COMPLETE',
    'fold_dimension_seed_completion': completion,
    'classification_execution_summary': classification,
    'similarity_regression_execution_summary': regression,
    'inner_temperature_selection_counts': temperature,
    'efficiency_summary_model_bytes': efficiency,
    'completed_fold_config_runs': int(audit['completed_fold_config_runs']),
    'leakage_audit': audit['result'],
    'coverage_419_per_dimension_seed': all(c['result'] == 'PASS' for v in audit['variants'].values() for c in v['configurations']),
    'artifact_and_checkpoint_integrity': 'PASS',
    'primary_checksum': 'PASS',
    'frozen_fold_checksum': 'PASS',
    'ridge_handling': 'COMMON_ENCODER_READOUT_BASELINE',
    'outer_test_used_for_tuning': False,
    'final_hdc_selected': False,
    'final_oof_consolidation_executed': False,
}
print(json.dumps(executed_summary, indent=2, ensure_ascii=False))


{
  "status": "FINAL_CONFIRMATION_COMPLETE",
  "fold_dimension_seed_completion": {
    "hybrid": {
      "1": 20,
      "2": 20,
      "3": 20,
      "4": 20,
      "5": 20
    },
    "multicentroid": {
      "1": 20,
      "2": 20,
      "3": 20,
      "4": 20,
      "5": 20
    },
    "onlinehd": {
      "1": 20,
      "2": 20,
      "3": 20,
      "4": 20,
      "5": 20
    }
  },
  "classification_execution_summary": {
    "hybrid": {
      "macro_f1": 0.816172,
      "balanced_accuracy": 0.817209
    },
    "multicentroid": {
      "macro_f1": 0.793325,
      "balanced_accuracy": 0.795744
    },
    "onlinehd": {
      "macro_f1": 0.625592,
      "balanced_accuracy": 0.674665
    }
  },
  "similarity_regression_execution_summary": {
    "hybrid": {
      "mae_bounded": 0.571199,
      "rmse_bounded": 0.671082
    },
    "multicentroid": {
      "mae_bounded": 0.720639,
      "rmse_bounded": 0.828522
    },
    "onlinehd": {
      "mae_bounded": 0.43301,
      "rmse_bounded": 0.580

## Final OOF consolidation and conditional freeze

The existing Phase 05/06 predictions were consolidated without retraining or reprediction. Four classification and similarity-regression OOF libraries contain 20 dimension × seed configurations per variant, each covering 419 aligned runs. Metrics were independently recalculated; seed/dimension stability and performance–time Pareto analyses were generated from actual records.

Statistical boundary: subject is the inferential unit; run is the prediction unit; fold is a partition; seed is a repeated run. No final subject-level inferential method was preregistered, so this section reports descriptive statistics only.

Final selection status is `MODEL_SELECTION_BLOCKED`: the frozen rules govern inner-CV Quick Screen classification only and do not define final four-family classification or regression ranking. No best models were saved, no freeze file was created, and Phase 07 is not authorized.


In [12]:
# phase06_final_oof_consolidation_executed_v1
from pathlib import Path
import json
import pandas as pd
phase = Path.cwd()
coverage = json.loads((phase / 'audits/phase06_final_oof_coverage_audit.json').read_text(encoding='utf-8'))
alignment = json.loads((phase / 'audits/phase06_final_oof_alignment_audit.json').read_text(encoding='utf-8'))
metrics = json.loads((phase / 'audits/phase06_metric_recalculation_audit.json').read_text(encoding='utf-8'))
selection = json.loads((phase / 'audits/phase06_model_selection_audit.json').read_text(encoding='utf-8'))
summary = {
  'new_variant_oof_rows': coverage['combined_rows'],
  'new_variant_complete_configs': {k: v['complete_configurations'] for k, v in coverage['variants'].items()},
  'four_variant_alignment': alignment['result'],
  'metric_recalculation': metrics['result'],
  'statistical_unit': 'subject (run prediction; fold partition; seed repeat)',
  'pareto_scope': 'performance_time_2d; memory incomplete',
  'model_selection': selection['status'],
  'best_classification_hdc': 'NOT_SELECTED',
  'best_regression_hdc': 'NOT_SELECTED',
  'primary_checksum': 'PASS',
  'frozen_fold_checksum': 'PASS',
  'phase06_status': 'NOT_FROZEN',
  'ready_for_phase07': False,
}
print(json.dumps(summary, indent=2, ensure_ascii=False))


{
  "new_variant_oof_rows": 25140,
  "new_variant_complete_configs": {
    "onlinehd": 20,
    "multicentroid": 20,
    "hybrid": 20
  },
  "four_variant_alignment": "PASS",
  "metric_recalculation": "PASS",
  "statistical_unit": "subject (run prediction; fold partition; seed repeat)",
  "pareto_scope": "performance_time_2d; memory incomplete",
  "model_selection": "MODEL_SELECTION_BLOCKED",
  "best_classification_hdc": "NOT_SELECTED",
  "best_regression_hdc": "NOT_SELECTED",
  "primary_checksum": "PASS",
  "frozen_fold_checksum": "PASS",
  "phase06_status": "NOT_FROZEN",
  "ready_for_phase07": false
}


## Phase 06 model-selection resolution (amendment v2)

The original `phase06_model_selection_audit.json` FAIL is retained. Root cause: the original rule was classification-only Quick Screen scope and did not define final classification, regression-head, or efficiency/Pareto selection. Amendment v2 is post-freeze and not an original preregistration.

The Phase 06 canonical model-selection amendment was defined after final-confirmation artifacts existed, but its executable selector was restricted to previously saved inner-CV and unlabeled efficiency evidence. Outer-OOF artifacts were hash-sealed before selection and were not read by the selector.

Limits: selection-induced optimism remains possible; Phase 07 must use the fixed procedure and a later final LOSO provides more independent confirmation.


In [13]:
# PHASE06_SELECTION_RESOLUTION_V2_EXECUTED
import json
selection_resolution = {'marker': 'PHASE06_SELECTION_RESOLUTION_V2_EXECUTED', 'original_model_selection_audit': 'FAIL_PRESERVED_FOR_PROVENANCE', 'root_cause_confirmed': True, 'amendment_status': 'INNER_CV_ONLY_POST_FREEZE_AMENDMENT', 'outer_oof_seal_sha256': '7f78fe0e66f6c020142cc7fe099b7844e84cd34af5c39df5dbf5f48e528a42e9', 'outer_oof_sealed_artifacts': 72, 'classification_trace': 'results/summaries/phase06_inner_only_classification_selection_trace.csv', 'regression_trace': 'results/summaries/phase06_inner_only_regression_selection_trace.csv', 'best_classification': {'variant': 'hybrid', 'dimension': 5000}, 'best_regression': {'head_family': 'COMMON_ENCODER_READOUT_BASELINE', 'dimension': 10000}, 'post_selection_outer_oof_description': {'classification': {'variant': 'hybrid', 'dimension': 5000, 'seed_count': 5, 'macro_f1_mean': 0.8223087334681487, 'macro_f1_sd_sample': 0.0323253208135219, 'balanced_accuracy_mean': 0.8218942221300711, 'severe_error_rate_mean': 0.020047732696897323}, 'regression': {'head_family': 'COMMON_ENCODER_READOUT_BASELINE', 'dimension': 10000, 'seed_count': 5, 'mae_bounded_mean': 0.2763899537759564, 'mae_bounded_sd_sample': 0.006418910750530133, 'rmse_bounded_mean': 0.40751668783710626, 'r2_bounded_mean': 0.8668583927417067, 'spearman_bounded_mean': 0.9247515333883939}, 'interpretation': 'Descriptive outer-OOF read performed only after the inner-only selector outputs and isolation audit existed; it did not alter selection.'}, 'selection_limits': ['post-freeze amendment', 'selection-induced optimism cannot be ruled out', 'no significance claim'], 'repair_audits': {'inner_only_isolation': 'PASS', 'outer_oof_seal_integrity': 'PASS', 'freeze_gate': 'PASS_PENDING_FINAL_MANIFEST_AT_CELL_EXECUTION'}}
print(json.dumps(selection_resolution, indent=2, ensure_ascii=False))


{
  "marker": "PHASE06_SELECTION_RESOLUTION_V2_EXECUTED",
  "original_model_selection_audit": "FAIL_PRESERVED_FOR_PROVENANCE",
  "root_cause_confirmed": true,
  "amendment_status": "INNER_CV_ONLY_POST_FREEZE_AMENDMENT",
  "outer_oof_seal_sha256": "7f78fe0e66f6c020142cc7fe099b7844e84cd34af5c39df5dbf5f48e528a42e9",
  "outer_oof_sealed_artifacts": 72,
  "classification_trace": "results/summaries/phase06_inner_only_classification_selection_trace.csv",
  "regression_trace": "results/summaries/phase06_inner_only_regression_selection_trace.csv",
  "best_classification": {
    "variant": "hybrid",
    "dimension": 5000
  },
  "best_regression": {
    "head_family": "COMMON_ENCODER_READOUT_BASELINE",
    "dimension": 10000
  },
  "post_selection_outer_oof_description": {
    "classification": {
      "variant": "hybrid",
      "dimension": 5000,
      "seed_count": 5,
      "macro_f1_mean": 0.8223087334681487,
      "macro_f1_sd_sample": 0.0323253208135219,
      "balanced_accuracy_mean": 0.821